app/src/main/java/ml/docilealligator/infinityforreddit/markdown/SpoilerParserPlugin.java

```diff
@@ -159,28 +155,57 @@ private ArrayList<Pair<Integer, Integer>> parse(SpannableStringBuilder markdown,
                     && (currentChar != '<')
                     && (currentChar != '!')) {
                 new_lines = 0;
-            } else if ((i + 1 < length)
-                    && currentChar == '>'
-                    && markdown.charAt(i + 1) == '!') {
-                openSpoilerStack.push(i + 2);
-            } else if ((i + 1 < length) && (i - 1 >= 0)
-                    && openSpoilerStack.size() > 0
-                    && markdown.charAt(i - 1) != '>'
+            } else if (currentChar == '>'
+                    && i + 1 < length
+                    && markdown.charAt(i + 1) == '!'
+                    && noCodeIntersection(markdown, i)) {
```

Comment:
Code block intersection detection must be here to ignore spoiler tags completely. Fixes
```
>! one `two !<` three !< - doesn't show any spoilers, should be one big spoiler
```